h/groups/views.py

```diff
@@ -101,26 +103,22 @@ def _read_group(request, group):
     """
     url = request.route_url('group_read', pubid=group.pubid, slug=group.slug)
 
-    result = search.search(request,
-                           private=False,
-                           params={"group": group.pubid, "limit": 1000})
-    annotations = [presenters.AnnotationHTMLPresenter(
-                       storage.annotation_from_dict(a))
-                   for a in result['rows']]
-
-    # Group the annotations by URI.
-    # Create a dict mapping the (normalized) URIs of the annotated documents
-    # to the most recent annotation of each document.
-    annotations_by_uri = collections.OrderedDict()
-    for annotation in annotations:
-        normalized_uri = uri.normalize(annotation.uri)
-        if normalized_uri not in annotations_by_uri:
-            annotations_by_uri[normalized_uri] = annotation
-            if len(annotations_by_uri) >= 25:
-                break
-
-    document_links = [annotation.document_link
-                      for annotation in annotations_by_uri.values()]
+    annotations = (
+        request.db.query(models.Annotation)
+        .options(
+            joinedload(models.Annotation.document)
+            .joinedload(models.Document.meta))
+        .filter(models.Annotation.shared == expression.true())
```

Comment:
Oh yeah, I did have more than one class in the `request.db.query(models.Annotation)` bit at one point which is why it seemed like I had to use `filter()`. It can be less verbose now